doc/source/configuration/vault.rst

```diff
@@ -0,0 +1,207 @@
+
+2. Install the Python package (with the Kayobe virtualenv activated)
+
+.. code-block::
+
+   pip install -r requirements.txt
+
+Clone the StackHPC Hashicorp Vault collection
+---------------------------------------------
+
+1. Add the following into the kayobe-config/etc/kayobe/ansible/requirements.yml
+
+.. code-block::
+
+   collections:
+     - name: stackhpc.hashicorp
+       version: 2.3.0
+
+2. Perform a control host upgrade to pull down the collection
+
+.. code-block::
+
+   kayobe control host upgrade
+
+Setup Vault on the seed node
+----------------------------
+
+1. Run vault-deploy-seed.yml custom playbook
+
+.. code-block::
+
+   kayobe playbook run ansible/vault-deploy-seed.yml
+
+2. Encrypt generated certs/keys with ansible-vault (use proper location of vault password file)
+
+.. code-block::
+
+   ansible-vault encrypt --vault-password-file ~/vault.pass vault/OS-TLS-INT.pem
+   ansible-vault encrypt --vault-password-file ~/vault.pass vault/seed-vault-keys.json
+   ansible-vault encrypt --vault-password-file ~/vault.pass vault/overcloud.key
+
+Setup HAProxy config for Vault
+------------------------------
+
+1. Create the HAProxy config to reverse proxy the Vault HA container
+
+Set the vault_front to the external VIP address or internal VIP address depending on the installation. Set the vault_back to the IPs of the control nodes.
+Set the following in etc/kayobe/kolla/config/haproxy/services.d/vault.cfg or if environments are being used etc/kayobe/environments/$KAYOBE_ENVIRONMENT/kolla/config/haproxy/services.d/vault.cfg
+
+.. code-block::
+
+   # Delete "verify none" if not using self-signed/unknown issuer
+   {% raw %}
+   frontend vault_front
+      mode tcp
+      option tcplog
+      bind {{ kolla_internal_vip_address }}:8200
+      default_backend vault_back
+
+   backend vault_back
+      mode tcp
+      option httpchk GET /v1/sys/health
+      # https://www.vaultproject.io/api-docs/system/health
+      # 200: initialized, unsealed, and active
+      # 501: not initialised (required for bootstrapping)
+      # 503: sealed (required for bootstrapping)
+      http-check expect rstatus (200|501|503)
+
+   {% for host in groups['control'] %}
+   {% set host_name = hostvars[host].ansible_facts.hostname %}
+   {% set host_ip = 'api' | kolla_address(host) %}
+      server {{ host_name }} {{ host_ip }}:8200 check check-ssl verify none inter 2000 rise 2 fall 5
+   {% endfor %}
+   {% endraw %}
+
+2. Deploy HAProxy with the new Vault service configuration:
+
+.. code-block::
+
+   kayobe overcloud service deploy -kt haproxy
+
+Setup Vault HA on the overcloud hosts
+-------------------------------------
+
+1. Run vault-deploy-overcloud.yml custom playbook
+
+.. code-block::
+
+   kayobe playbook run ansible/vault-deploy-overcloud.yml
+
+2. Encrypt overcloud vault keys (use proper location of vault password file)
+
+.. code-block::
+
+   ansible-vault encrypt --vault-password-file ~/vault.pass vault/overcloud-vault-keys.json
+
```

Comment:
Perhaps add another top level section here to delineate deployment of vault from certificate generation?